x/cron/keeper/keeper.go

```diff
@@ -67,19 +67,32 @@ func (k *Keeper) Logger(ctx sdk.Context) log.Logger {
 // ExecuteReadySchedules gets all schedules that are due for execution (with limit that is equal to Params.Limit)
 // and executes messages in each one
 // NOTE that errors in contract calls rollback all already executed messages
-func (k *Keeper) ExecuteReadySchedules(ctx sdk.Context) {
+func (k *Keeper) ExecuteReadySchedules(ctx sdk.Context, isBeginBlocker bool) {
```

Comment:
Looks like `isBeginBlocker` arg can be or type `Blocker` (until renamed of course). That way we can directly compare schedule blocker param with the passed arg

Reply:
Also there are two checks but can be only one: 
How about refactoring it something like this (NOTE: not tested!)
```go
func (k *Keeper) ExecuteReadySchedules(ctx sdk.Context, currentStage types.BlockerType) {
	telemetry.ModuleMeasureSince(types.ModuleName, time.Now(), LabelExecuteReadySchedules)
	schedules := k.getSchedulesReadyForExecution(ctx)

	for _, schedule := range schedules {
		if IsExecutableAtThisStage(schedule, currentStage) {
			err := k.executeSchedule(ctx, schedule)
			recordExecutedSchedule(err, schedule)
		}
	}
}

func IsExecutableAtThisStage(schedule types.Schedule, currentStage types.BlockerType) bool {
	return schedule.Blocker == types.BlockerType_BOTH || schedule.Blocker == currentStage
}

```

Reply:
Done